test/distributed/_tensor/test_math_ops.py

```diff
@@ -393,6 +421,147 @@ def _replicate_fn(name, module, device_mesh):
 
             self.assertEqual(x_local.grad, x_dist.grad.full_tensor())
 
+    @with_comms
+    def test_layer_norm_bwd_req_grad(self):
+        device_mesh = self.build_device_mesh()
+        batch, seq_len, embedding_dim, vocab_size = 8, 8, 10, 32
+
+        # build our subtest configurations and filter out invalid ones
+        class SubTest(NamedTuple):
+            multidim_norm: bool
+            elementwise_affine: bool
+            emb_req_grad: bool
+            ln_req_grad: bool
+            out_req_grad: bool
+
+        subtest_fails = {}
+        valid_filter = lambda cfg: not (  # noqa: E731
+            cfg.ln_req_grad and not cfg.elementwise_affine
+        ) and any(cfg[2:])
+        subtest_cfgs = list(
+            filter(
+                valid_filter,
+                [SubTest(*cfg) for cfg in itertools.product(*(((False, True),) * 5))],
+            )
+        )
+
+        for subtest_cfg in subtest_cfgs:
+            try:
+                (
+                    multidim_norm,
+                    elementwise_affine,
+                    emb_req_grad,
+                    ln_req_grad,
+                    out_req_grad,
+                ) = subtest_cfg
+                normalized_shape = (
+                    (seq_len, embedding_dim) if multidim_norm else (embedding_dim,)
+                )
+
+                # configure our local and parallelized models for this subtest
+                class LnTpBlock(torch.nn.Module):
+                    def __init__(self):
+                        super().__init__()
+                        self.preln_embeddings = torch.nn.Embedding(
+                            vocab_size, embedding_dim
+                        )
+                        self.layer_norm = torch.nn.LayerNorm(
+                            normalized_shape, elementwise_affine=elementwise_affine
+                        )
+                        self.postln_linear = torch.nn.Linear(
+                            embedding_dim, embedding_dim
+                        )
+
+                    def forward(self, tokens):
+                        h = self.preln_embeddings(tokens)
+                        h = self.layer_norm(h)
+                        output = self.postln_linear(h)
+                        return output
+
+                parallel_plan = {
+                    "preln_embeddings": RowwiseParallel(
+                        input_layouts=Replicate(), output_layouts=Shard(1)
+                    ),
+                    "layer_norm": SequenceParallel(),
+                    "postln_linear": ColwiseParallel(
+                        input_layouts=Shard(1),
+                        output_layouts=Replicate(),
+                    ),
+                }
+
+                model = LnTpBlock().to(device=self.device_type)
```

Comment:
this line can be safely removed since `parallelize_module` will automatically move the parameters to device. See: https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/api.py#L407-L408

Reply:
Good point! Updated.